applications/TrilinosApplication/custom_strategies/convergencecriterias/trilinos_mixed_generic_criteria.h

```diff
@@ -0,0 +1,281 @@
+    ///@name Type Definitions
+    ///@{
+
+    KRATOS_CLASS_POINTER_DEFINITION(TrilinosMixedGenericCriteria);
+
+    typedef MixedGenericCriteria< TSparseSpace, TDenseSpace > BaseType;
+
+    typedef typename BaseType::TDataType TDataType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+
+    typedef typename BaseType::ConvergenceVariableListType ConvergenceVariableListType;
+
+    typedef typename BaseType::KeyType KeyType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+
+    /**
+     * @brief Construct a new Trilinos Mixed Generic Criteria object
+     * Construct the mixed generic convergence criteria from a convergence variables list.
+     * The convergence variable list contains for each variable the variable itself as well as the corresponding relative and absolute tolerances.
+     * @param rConvergenceVariablesList List containing tuples with the convergence variables to be checked. The tuples are set as <Variable, relative tolerance, absolute tolerance>
+     */
+    TrilinosMixedGenericCriteria(const ConvergenceVariableListType& rConvergenceVariablesList)
+        : MixedGenericCriteria<TSparseSpace, TDenseSpace>(rConvergenceVariablesList)
+    {}
+
+    /// Destructor.
+    ~TrilinosMixedGenericCriteria() override
+    {}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Compute relative and absoute error.
+    /**
+     * @param rModelPart Reference to the ModelPart containing the fluid problem.
+     * @param rDofSet Reference to the container of the problem's degrees of freedom (stored by the BuilderAndSolver)
+     * @param A System matrix (unused)
+     * @param Dx Vector of results (variations on nodal variables)
+     * @param b RHS vector (residual)
+     * @return true if convergence is achieved, false otherwise
+     */
+    bool PostCriteria(
+        ModelPart& rModelPart,
+        DofsArrayType& rDofSet,
+        const TSystemMatrixType& A,
+        const TSystemVectorType& Dx,
+        const TSystemVectorType& b) override
+    {
+        // If required, initialize the Epetra vector import
+        if(!mEpetraImportIsInitialized) {
+            InitializeEpetraImport(rDofSet, Dx);
+        }
+
+        // Check if we are solving for something
+        if (TSparseSpace::Size(Dx) != 0) {
+            // Calculate the convergence ratio and absolute norms
+            const auto convergence_norms = BaseType::CalculateConvergenceNorms(rModelPart, rDofSet, Dx);
+
+            // Output convergence status
+            BaseType::OutputConvergenceStatus(convergence_norms);
+
+            // Check convergence
+            return BaseType::CheckConvergence(convergence_norms);
+        } else {
+            // Case in which all the DOFs are constrained!
+            return true;
+        }
```

Comment:
```suggestion
        BaseType::PostCriteria(...)
```

Reply:
I did it in that way so it can be easily followed how the convergence is checked, but I agree it has not so much sense. Changed.